src/vm/dllimport.cpp

```diff
@@ -7178,6 +7197,50 @@ HINSTANCE NDirect::LoadLibraryModule( NDirectMethodDesc * pMD, LoadLibErrorTrack
         }
     }
 
+#ifdef FEATURE_PAL
+    if (hmod == NULL)
+    {
+        // P/Invokes are often declared with variations on the actual library name.
+        // For example, it's common to leave off the extension/suffix of the library
+        // even if it has one, or to leave off a prefix like "lib" even if it has one
+        // (both of these are typically done to smooth over cross-platform differences). 
+        // We try to dlopen with such variations on the original.
+        const char* const prefixSuffixCombinations[] =
+        {
+            "%s%s%s",     // prefix+name+suffix
+            "%.0s%s%s",   // name+suffix
+            "%s%s%.0s",   // prefix+name
+        };
+
+        const int NUMBER_OF_LIB_NAME_VARIATIONS = COUNTOF(prefixSuffixCombinations);
+        SString libNameVariations[NUMBER_OF_LIB_NAME_VARIATIONS];
```

Comment:
Yup, this is leftover from the previous iteration, but is no longer needed. I'll remove it.